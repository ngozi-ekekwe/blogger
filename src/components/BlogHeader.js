import React, { Component } from 'react';
import { connect } from 'react-redux';
import Profile from './Profile';

class BlogHeader extends Component {
  render() {
    const { story } = this.props;
    return (
      <div className="row blog-section">
        <div className="col-12">
          <div className="blog-content">
            {
              story[0] &&<h1>{story[0].title}</h1>
            }
            { story[0]  && <Profile userId={story[0].userId }/>}
          </div>
        </div>
        <div className="col-12 blog-image"></div>
      </div>
    );
  }
}

export default BlogHeader;