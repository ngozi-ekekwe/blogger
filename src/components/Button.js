import React from 'react';

const Button = ({ title, action, variant, onClick }) => {
  return (
    <button className={`btn btn-${variant}`} onClick={onClick}>
      {title}
    </button>
  );
};

export default Button;