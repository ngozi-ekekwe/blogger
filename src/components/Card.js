import React from 'react';
import Profile from '../components/Profile'
import { Link } from '../routes';

const Card = (props) => {
  const { title, content, slug } = props;
  return (
    <Link href={`/blog/${slug}`}>
      <div className="col-12 col-lg-6 card">
        <div className="row">
          <div className="col-8">
            <h2 className="card-title">{title}</h2>
            <p className="card-subtitle">{content} </p>

            <Profile />
          </div>
          <div className="col-4 card-image"></div>
        </div>
      </div>
    </Link>
  );
};

export default Card;