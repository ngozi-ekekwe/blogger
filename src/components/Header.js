import React, { Component } from 'react';
import { Link } from '../routes';
import '../styles/style.scss';

class Header extends Component {
  state = {
    isAuthenticated: false
  }
  
  logout = () => {
    const token = localStorage.removeItem('token');
  }

  componentDidMount() {
    const token = localStorage.getItem('token')
    if(token) {
      this.setState({
        isAuthenticated: true
      })
    }
  }
  render() {
    const { isAuthenticated } = this.state;
    return (
      <header>
        <div className="nav-brand">
          <Link route='/home'>STORY</Link>
        </div>
        {!isAuthenticated && <nav>
          <ul className="nav-item">
            <Link route='/login'><a>Sign in</a></Link>
          </ul>
          <ul className="nav-item active">
            <Link route='/signup'><a>Get Started</a></Link>
          </ul>
        </nav>}
        {isAuthenticated && <nav>
          <ul className="nav-item">
            <Link route='/new-story'><a>Create</a></Link>
          </ul>
          <ul className="nav-item">
            <Link route='/blogs'><a>Stories</a></Link>
          </ul>
          <ul className="nav-item active">
            <Link route='/signup'><a onClick={this.logout}>Logout</a></Link>
          </ul>
          <Link>
          <div className="avatar ml-5"></div>
        </Link>
        </nav>}
      </header>
    );
  }
}

export default Header;