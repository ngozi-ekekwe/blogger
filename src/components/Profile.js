import React, { Component } from 'react';
import readingTime from 'reading-time';
import moment from 'moment';
import { connect } from 'react-redux';
import { Link } from '../routes';

export class Profile extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const { userId, getUser } = this.props;
    userId && getUser(userId)
  }

  render() {
    const { user, content={} } = this.props;
    return (
      <div className="user-profile">
        <div className="flex-0">
          <Link>
            <div className="avatar"></div>
          </Link>
        </div>
       { user && <div className="flex-1 profile-details">
            <p className="profile-name">{`${user.firstName} ${user.lastName}`}</p>
            <div className="date">{`${moment(user.createdAt).format('MMMM Do YYYY')}  - ${readingTime(content).text}`}</div>
        </div>}
      </div>
    );
  }
}

function mapStateToProps(state, props) {
  return {
    user: state.userReducer.user[0]
  };
}

const mapDispatchToProps = (dispatch) => {
  return ({
    getUser: (userId) => {
      return dispatch({ type: 'GET_USER', userId })
    }
  });
}

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
