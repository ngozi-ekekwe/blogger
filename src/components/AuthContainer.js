import React, { Component } from 'react';
import Button from './Button';

const AuthContainer = (
  { title, buttonText, children, onButtonClick, loading }
) => {
  return (
    <div className="row authentication-section">
      <div className="col-12 col-lg-6 authentication-image"></div>
      <div className="col-12 col-lg-6">
        <div className="authentication-content">
          <h1>{title}</h1>
          {children}
        </div>
        <div className="button-container">
          <Button title={buttonText} onClick={onButtonClick} loading={loading} />  
        </div>       
      </div>
    </div>
  );
}

export default AuthContainer;