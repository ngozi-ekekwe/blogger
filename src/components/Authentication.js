import React, { Component } from 'react';
import Button from './Button';

class Authentication extends Component {
  constructor(props){
    super(props);
  }

  render() {
    const { title, buttonText, children, onButtonClick } = this.props;
    return (
      <div className="row authentication-section">
        <div className="col-12 col-lg-6 authentication-image"></div>
        <div className="col-12 col-lg-6">
          <div className="authentication-content">
            <h1>{title}</h1>
            {children}
          </div>
          <div className="button-container">
            <Button title={buttonText} onClick={onButtonClick} />   
          </div>       
        </div>
      </div>
    );
  }
}

export default Authentication;