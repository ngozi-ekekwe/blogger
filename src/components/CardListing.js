import React from 'react';
import Card from './Card';

const CardListing = ({stories}) => {
  return (
    <div className="row card-listing">
      { stories && stories.map((story) => {
        return <Card  title={story.title} content={story.content}/>
      })}
    </div>
  );
};

export default CardListing;