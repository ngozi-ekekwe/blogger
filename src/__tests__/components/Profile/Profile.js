import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Profile } from '../../../components/Profile';

describe('Profile', () => {
  describe('renders the component', () => {
    test('it renders the component correctly', () => {
      const component = shallow(<Profile user={1} content={'Hello world'} />);
      expect(toJson(component)).toMatchSnapshot();
    })
  })


  describe('Get the users profile', () => {
    test('', () => {
      const component = shallow(<Profile user={1} content={'Hello world'} dispatch={jest.fn()} />);
      component.update();
      expect(toJson(component)).toMatchSnapshot();

    })
  })
})