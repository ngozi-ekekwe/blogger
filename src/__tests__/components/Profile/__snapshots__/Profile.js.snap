// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Get the users profile  1`] = `
<div
  className="user-profile"
>
  <div
    className="flex-0"
  >
    <LinkRoutes>
      <div
        className="avatar"
      />
    </LinkRoutes>
  </div>
  <div
    className="flex-1 profile-details"
  >
    <p
      className="profile-name"
    >
      undefined undefined
    </p>
    <div
      className="date"
    >
      March 19th 2019  - 1 min read
    </div>
  </div>
</div>
`;

exports[`Profile renders the component it renders the component correctly 1`] = `
<div
  className="user-profile"
>
  <div
    className="flex-0"
  >
    <LinkRoutes>
      <div
        className="avatar"
      />
    </LinkRoutes>
  </div>
  <div
    className="flex-1 profile-details"
  >
    <p
      className="profile-name"
    >
      undefined undefined
    </p>
    <div
      className="date"
    >
      March 19th 2019  - 1 min read
    </div>
  </div>
</div>
`;
